src/mono/dbi/cordb-process.cpp

```diff
@@ -0,0 +1,657 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+// File: CORDB-PROCESS.CPP
+//
+
+#include <cordb-appdomain.h>
+#include <cordb-assembly.h>
+#include <cordb-breakpoint.h>
+#include <cordb-code.h>
+#include <cordb-eval.h>
+#include <cordb-function.h>
+#include <cordb-process.h>
+#include <cordb-thread.h>
+#include <cordb.h>
+
+using namespace std;
+
+CordbProcess::CordbProcess(Cordb* cordb) : CordbBaseMono(NULL)
+{
+    m_pAppDomainEnum = NULL;
+    m_pBreakpoints   = new ArrayList();
+    m_pThreads       = new ArrayList();
+    m_pFunctions     = new ArrayList();
+    m_pModules       = new ArrayList();
+    appDomains       = new ArrayList();
+    m_pPendingEval   = new ArrayList();
+    this->m_pCordb   = cordb;
+    m_bIsJustMyCode  = false;
+}
+
+CordbProcess::~CordbProcess()
+{
+    if (m_pAppDomainEnum)
+        m_pAppDomainEnum->InternalRelease();
+    DWORD i = 0;
+    while (i < m_pBreakpoints->GetCount())
```

Comment:
Is there any reason why not use `for` statement ?

Reply:
no...

Reply:
Changed.